tests/unit/rbtree/main.c

```diff
@@ -245,9 +245,91 @@ void test_rbtree_spam(void)
 	} while (size < MAX_NODES);
 }
 
+/**
+ * @brief Test removing a node with abnormal color.
+ *
+ * @details Initialize a tree and insert it, and then modify
+ * the node's color manually forcely. And test the remove
+ * operations' behavor.
+ *
+ * @ingroup lib_rbtree_tests
+ *
+ * @see rb_get_min(), rb_get_max(), rb_remove()
+ */
+void test_rb_remove_black_err(void)
+{
+	struct rbtree tree;
+	struct rbnode nodes[5];
+
+	/* Initialize a tree and insert it */
+	(void)memset(&tree, 0, sizeof(tree));
+	tree.lessthan_fn = node_lessthan;
+	(void)memset(nodes, 0, sizeof(nodes));
+	/* Try to get min node, nil will be returned */
+	zassert_true(rb_get_min(&tree) == NULL, "the new tree is not empty");
+
+	for (int i = 0; i < 4; i++) {
+		rb_insert(&tree, &nodes[i]);
+	}
+
+	/* Check if tree's max and min node are expected */
+	zassert_true(rb_get_min(&tree) == &nodes[0], "the tree is incorrect");
+	zassert_true(rb_get_max(&tree) == &nodes[3], "the tree is incorrect");
+
+	/* Try to remove a node that not in the tree will be returned directly */
+	rb_remove(&tree, &nodes[4]);
+
+	/* Manually set the color of the node to make the tree abnormal,
```

Comment:
Hi @andyross , yep, I designed these two testcase is to test for coverage.
And see the picture as below:
![image](https://user-images.githubusercontent.com/62736294/109740250-f3afd680-7c05-11eb-9820-d4a0e71d2776.png)
I tried to remove a node that not belong to the tree, this means it will cannot find this node in find_and_stack function, and will be returned directly. And function rb_remove don't have return value, so I check coverage report to verify this condition is successful or failed.
And another picture as below:
![image](https://user-images.githubusercontent.com/62736294/109741061-4e95fd80-7c07-11eb-8ced-63231f3f2dfa.png)
As you see, this part is to check color of a tree, if a tree is valid, it will conform the balance-tree's color principle. But it's difficult to design a invalid tree, so I changed color of the tree manually to make it invalid. After this, the source code will check and fix missing black.
So Andy, do you think this works for testing? Or is there a better way to test this instead?